app/controllers/api/v1/accounts/statuses_controller.rb

```diff
@@ -31,6 +31,7 @@ def account_statuses
     default_statuses.tap do |statuses|
       statuses.merge!(only_media_scope) if params[:only_media]
       statuses.merge!(no_replies_scope) if params[:exclude_replies]
+      statuses.merge!(filter_with_visibility_scope(params[:visibility])) if params[:visibility]
```

Comment:
When anonymous viewer requests this, the timeline's default filter is "WHERE visibility IN (public, unlisted)", doesn't this essentially replace that, allowing access to "WHERE visibility = private"? This is the main reason I haven't merged this PR, and I don't think I should. This messes with access controls.